src/app/utilities/redirect.js

```diff
@@ -1,43 +1,55 @@
-import { fromPairs } from "lodash";
 import { browserHistory } from "react-router";
 import { store } from "../config/store";
-import fp from "lodash/fp";
-
 /**
  * Redirects to one of the main views in Florence - chooses whether it needs to redirect to old Florence or route within the new application
  *
  * @param {string} - The path of the view that we want to redirect to
  */
 
-export default function redirectToMainScreen(screen) {
+export default function handleRedirect(redirectPath) {
+    if (!redirectPath) {
+        return internalRedirect();
+    }
+
+    const config = window.getEnv();
+    const allowedExternalPaths = config.allowedExternalPaths;
+    if (allowedExternalPaths.some(path => redirectPath.startsWith(path))) {
+        return externalRedirect(redirectPath);
+    }
+
+    internalRedirect(redirectPath);
```

Comment:
Why no return?

Reply:
hmm git copilot deemed it unnecessary when I checked my refactoring, will put it back in

Reply:
Done